aspects/aspects.go

```diff
@@ -441,10 +441,39 @@ type expandedMatch struct {
 	value interface{}
 }
 
+// validateSetValue checks that map keys conform to the same format a path sub-keys.
+// TODO: check recursion limit here as well
+func validateSetValue(v interface{}) error {
+	var nestedVals []interface{}
+	switch typedVal := v.(type) {
+	case map[string]interface{}:
+		for k, v := range typedVal {
+			if !validSubkey.Match([]byte(k)) {
+				return fmt.Errorf(`key %q doesn't conform to required format: %s`, k, validSubkey.String())
+			}
+
+			nestedVals = append(nestedVals, v)
+		}
+
+	case []interface{}:
+		nestedVals = typedVal
+	}
+
+	for _, v := range nestedVals {
+		if err := validateSetValue(v); err != nil {
+			return err
+		}
```

Comment:
Nested values recursion is not exercised in tests

Reply:
Thanks, I've expanded this now